ansible_collections/arista/avd/roles/eos_designs/python_modules/sflow/avdstructuredconfig.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.filter.natural_sort import natural_sort
+from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdMissingVariableError
+from ansible_collections.arista.avd.plugins.plugin_utils.strip_empties import strip_null_from_data
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+
+class AvdStructuredConfigSflow(AvdFacts):
+    """
+    This class must be rendered after all other eos_designs modules since it relies on
+    detecting sflow from the interface structured config generated by the other modules.
+
+    The only exception is of course custom_structured_configuration which always comes last.
+    """
+
+    @cached_property
+    def sflow(self) -> dict | None:
+        """
+        Structured config for sFlow.
+
+        Only configure if any interface is enabled for sFlow.
+
+        Covers:
+        - sflow_settings
+        - source-interfaces based on source_interfaces.sflow
+        """
+        if not self._enable_sflow:
+            return None
+
+        destinations = get(self._hostvars, "sflow_settings.destinations")
+        if destinations is None:
+            # TODO:
+            # AVD5.0.0 raise an error if sflow is enabled on an interface but there are no destinations configured.
+            # This cannot be implemented today since it would be breaking for already released support for sflow on interfaces.
```

Comment:
Feat(eos_designs)!: Raise an issue when Sflow enabled on an interface but no destination is configured

AVD5.0.0 raise an error if sflow is enabled on an interface but there are no destinations configured.
This cannot be implemented today since it would be breaking for already released support for sflow on interfaces.

Reply:
issue created. Thanks.